social_core/backends/openshift.py

```diff
@@ -0,0 +1,34 @@
+"""
+Openshift OAuth2 backend
+"""
+from six.moves.urllib.parse import urljoin
+from ..utils import append_slash
+from .oauth import BaseOAuth2
+import requests
+
+class OpenshiftOAuth2(BaseOAuth2):
+    name = 'openshift'
+    ACCESS_TOKEN_METHOD = 'POST'
+
+    def access_token_url(self):
+        return urljoin(append_slash(self.setting('URL')), 'oauth/token')
+
+    def authorization_url(self):
+        return urljoin(append_slash(self.setting('URL')), 'oauth/authorize')
+
+    def get_user_id(self, details, response):
+        return response['metadata']['uid']
+
+    def get_user_details(self, response):
+        """Return user details from openshift account"""
+        username = response['metadata']['name']
+        email = response['metadata']['name']
+        return {'username': username, 'email': email}
```

Comment:
using metadata name in both because openshift api does not return much information about user